DOC: Clarify advanced vs basic indexing tuple definition

muhammed-soysl opened the pull request:
The existing definition of advanced indexing was slightly confusing about what triggers it versus basic indexing when using a tuple.

This commit adds a note to explicitly state that a tuple containing *only* slices, integers, ellipsis, or newaxis triggers basic indexing, while a tuple containing *at least one* list or ndarray triggers advanced indexing.

Relates to #30022.

<!--         ----------------------------------------------------------------
                MAKE SURE YOUR PR GETS THE ATTENTION IT DESERVES!
                ----------------------------------------------------------------

*  FORMAT IT RIGHT:
      https://www.numpy.org/devdocs/dev/development_workflow.html#writing-the-commit-message

*  IF IT'S A NEW FEATURE OR API CHANGE, TEST THE WATERS:
      https://www.numpy.org/devdocs/dev/development_workflow.html#get-the-mailing-list-s-opinion

*  HIT ALL THE GUIDELINES:
      https://numpy.org/devdocs/dev/index.html#guidelines

*  WHAT TO DO IF WE HAVEN'T GOTTEN BACK TO YOU:
      https://www.numpy.org/devdocs/dev/development_workflow.html#getting-your-pr-reviewed
-->

jorenham commented on doc/source/user/basics.indexing.rst:
```diff
@@ -283,6 +283,17 @@ or a tuple with at least one sequence object or ndarray (of data type
+.. note::
+
+   The key difference from *basic* indexing lies in what the :class:`tuple`
+   contains. A tuple containing *only* integers, :class:`slice` objects,
```
It's not a bad suggestion actually, and also I agree that it's a nitpick, so feel free to ignore it if you disagree.

muhammed-soysl commented on doc/source/user/basics.indexing.rst:
```diff
@@ -283,6 +283,17 @@ or a tuple with at least one sequence object or ndarray (of data type
+
+   In contrast, a tuple containing at least one :class:`list`,
+   :class:`ndarray`, or other sequence object triggers
+   :ref:`advanced-indexing`.
```
Hi @jakevdp, thanks for taking the time to review this

My goal was to address the confusion (from Issue #30022) about what differentiates a "basic" tuple (which returns a view) from an "advanced" tuple (which returns a copy).

The original paragraph explained what *triggers* advanced indexing (a list/array in the tuple), but it didn't explicitly contrast it with what *triggers* basic indexing (a tuple with *only* integer scalars/slices). My note was an attempt to put those two rules side-by-side.

I agree that editing the original paragraph directly is a much better solution.

How about this: I can remove the `.. note::` block and instead modify the original paragraph to include this contrast? Something like:
"...This contrasts with *basic indexing* tuples, which contain *only* integer scalars, :class:`slice` objects, etc. An index tuple that contains *at least one* sequence object, :class:`ndarray`, or :class:`list` triggers :ref:`advanced-indexing`."

Let me know if an approach like that sounds better, and I'll update the PR.

muhammed-soysl committed:
DOC: Clarify definition of advanced vs basic indexing

Uses simpler logic, per jakevdp's review, to define advanced indexing as the case where an index is an array/sequence, and basic indexing as 'everything else'.

This directly resolves the confusion from #30022.

jorenham commented:
I'm sorry but I don't consider this to be an improvement, and I don't see this becoming one either, so I'm going to close this.
